doc/nrf/libraries/others/audio_module.rst

```diff
@@ -0,0 +1,120 @@
+It provides a common interface to audio processing algorithms.
+
+Using this interface, you can open and configure the custom modules, connect to them, and start and stop them.
+You can also send audio data to and from the application.
+
+The operation of the module is determined by a set of functions that perform the processing.
+Connecting these modules lets you create the following audio system or stream:
+
+.. figure:: images/audio_module_stream.svg
+   :alt: Audio stream example
+
+Implementation
+==============
+
+The audio module is implemented as a set of functions, listed in the following figure:
+
+.. figure:: images/audio_module_functions.svg
+   :alt: Audio module functions
+
+These functions call out to the user's implementation of these functions written to a predetermined API for the desired audio algorithm.
+
+The following table outlines the available functions that are defined in :c:struct:`audio_module_functions` and whether they are mandatory or not:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Function
+      - Mandatory/Optional
+      - Comment
+    * - ``audio_module_functions.*open``
+      - Optional
+      - Perform any algorithm operations to open.
+    * - ``audio_module_functions.*close``
+      - Optional
+      - Tidy up the algorithm on a close.
+    * - ``audio_module_functions.*configure_set``
+      - Mandatory
+      - Configure the algorithm to perform a particular operation.
+    * - ``audio_module_functions.*configuration_get``
+      - Mandatory
+      - Return the internal configuration of the algorithm.
+    * - ``audio_module_functions.*start``
+      - Optional
+      - Set the algorithm running.
+    * - ``audio_module_functions.*stop``
+      - Optional
+      - Stop the algorithms.
+    * - ``audio_module_functions.*data_process``
+      - Mandatory
+      - Process the data for the algorithm.
+
+The audio algorithm can run only if these functions are combined with the audio module.
+The audio module cannot perform a task on its own, as it merely supplies a consistent way to interface to an audio algorithm.
+
+The following figures show the flow and the internal states of the audio module:
+
+.. figure:: images/audio_module_flow.svg
+   :alt: Audio module flow
+
+.. figure:: images/audio_module_states.svg
+   :alt: Audio module internal states
+
+Configuration
+*************
+
+To use the audio module library, set the following Kconfig options to ``y`` in the project configuration file :file:`prj.conf`:
+
+* :kconfig:option:`CONFIG_AUDIO_MODULE`
+* :kconfig:option:`CONFIG_DATA_FIFO`
+
+Application integration
+***********************
+
+To create your own audio module for an LE Audio application, complete the following steps:
+
+#. Write the mandatory functions required by the function table API in `Implementation`_.
+#. Write any optional functions.
+#. Assign the function table to an instance of an audio module.
+#. Build with the audio module API and link together with the application.
```

Comment:
What is meant by "Build with the audio module API " here?

Reply:
This is the audio module common code, can we discuss wording?